arches/app/utils/decorators.py

```diff
@@ -59,6 +61,24 @@ def new_func(*args, **kwargs):
     return new_func
 
 
+def login_required(function):
+    @functools.wraps(function)
```

Comment:
@RobOatesHistoricEngland - Is this something has Farallon have used in their code?  If not, they might query it.  Just flagging this up as a possibility.

Reply:
Farallon have used decorators throughout the code to check/restrict access based on logged-in user permissions. There was nothing that redirected them back to the login screen if they were not logged-in. I added a little extra functionality so that API endpoints simply report a 403 Permission Denied, as it was not appropriate for an API endpoint to result in a 302 redirect to the login page.